### PR TITLE
Currency System

### PR DESCRIPTION
**Pull request by JuanGogi**
The currency class will allow the developers to add, deduct, save and  load the player money.

### CODE REVIEW

File: Assets/Scripts/Currency/CurrencySystem.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+using TMPro;
+
+public class CurrencySystem : MonoBehaviour
```
Comment (Gold3nEagle) [inline Assets/Scripts/Currency/CurrencySystem.cs:4]: the class should be a singleton

File: Assets/Scripts/Currency/CurrencySystem.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+
+    public int startingCurrency = 0;
+    public TextMeshProUGUI currencyText;
```
Comment (Gold3nEagle) [inline Assets/Scripts/Currency/CurrencySystem.cs:8]: the currency system should not be concerned/have a reference to anything outside of its jurisdiction.

so it should not have a reference to text gameobject.

instead, make a unity event that is triggered whenever there is a change to the currency

research unity events and the Observer Design Pattern

File: Assets/Scripts/Currency/CurrencySystem.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    private int currentCurrency;
```
Comment (Gold3nEagle) [inline Assets/Scripts/Currency/CurrencySystem.cs:10]: create a property for the currentCurrency to allow access only for the property outside of the class

research C# class properties.

File: Assets/Scripts/Currency/CurrencySystem.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        UpdateCurrencyText();
+    }
+
+    public void SubtractCurrency(int amount)
+    {
+        currentCurrency -= amount;
+        UpdateCurrencyText();
+    }
+
+    private void UpdateCurrencyText()
```
Comment (Gold3nEagle) [inline Assets/Scripts/Currency/CurrencySystem.cs:30]: once you have the Event, remove all references of the text and related functions like this one

**Comment by JuanGogi** [comment]
Check the fixes done

**Comment by Gold3nEagle** [comment]
Approved

**Pull request merged**